### PR TITLE
CAMEL-21358: Add Knative Http SSL client options

### PR DESCRIPTION
**Pull request by christophd**
# Description

- Provide Http client options to easily connect with SSL secured Knative broker
- Support keystore and truststore in different formats (jks, p12, pem)
- Support multiple key and trust certificates
- Support autoconfiguration of client options via system property and environment variable property binding
- Support bean lookup of client options via Camel context
- Improve Knative component docs
- Add Knative Http transport docs

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

See CAMEL-21358

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally from root folder and I have committed all auto-generated changes.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

File: components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc [hunk 1]
```diff
@@ -137,13 +137,21 @@ RouteBuilder.addRoutes(context, b -> {
 <1> set the location of the `Knative Environment` file
 <2> expose knative service
 
-=== Using custom Knative Transport
+You can also use property based component configuration to set the `Knative Environemnt` configuration file.
```
Comment (squakez) [inline components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc:140]:
```suggestion
You can also use property based component configuration to set the `Knative Environment` configuration file.
```

File: components/camel-knative/camel-knative-http/src/main/java/org/apache/camel/component/knative/http/TrustAllOptions.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.TrustManagerFactorySpi;
+import javax.net.ssl.X509TrustManager;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.net.TrustOptions;
+
+/**
+ * Trust options for trusting all client and server certificates. Do not use in production environments but primarily in
+ * test environments. Implementation is heavily based on the Quarkus TLS registry implementation.
```
Comment (squakez) [inline components/camel-knative/camel-knative-http/src/main/java/org/apache/camel/component/knative/http/TrustAllOptions.java:35]: How would this behave on main or Spring Boot runtimes?

File: components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc [hunk 3]
```diff
@@ -137,13 +137,21 @@ RouteBuilder.addRoutes(context, b -> {
...
 
-As today the component only support `http` as transport as it is the only supported protocol on Knative side but the transport is pluggable by implementing the following interface:
+.application.properties
+[source,properties]
+----
+camel.component.knative.environmentPath=classpath:knative.json
```
Comment (squakez) [inline components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc:145]: Is this anyhow changing any previous behavior. I am not into details of the implementations, just wondering if by any chance we'll be introducing any breaking compatibility. I recall we gave the possibility to initialize environment from an environment variable as well, but I may be making confusion.

File: components/camel-knative/camel-knative-http/src/main/java/org/apache/camel/component/knative/http/TrustAllOptions.java (hunk 2 above)
Comment (christophd) [reply components/camel-knative/camel-knative-http/src/main/java/org/apache/camel/component/knative/http/TrustAllOptions.java:35]: It is runtime agnostic and works also on main or SpringBoot. The implementation is just heavily inspired from Quarkus

File: components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc (hunk 1 above)
Comment (christophd) [reply components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc:140]: ups, nice catch!

File: components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc (hunk 3 above)
Comment (christophd) [reply components/camel-knative/camel-knative-component/src/main/docs/knative-component.adoc:145]: no, this is not a new behavior and it is also working with `CAMEL_COMPONENT_KNATIVE_ENVIRONMENT_PATH` envVar

**Pull request merged**